src/test/java/uk/gov/digital/ho/pttg/application/HmrcCallWrapperTest.java

```diff
@@ -142,4 +179,79 @@ public void shouldRethrowOtherExceptionsWithTraverson() {
                 }));
     }
 
+    @Test
+    public void shouldLogWhenInternalServerErrorException() {
+        given(mockRestTemplate.exchange(any(URI.class), any(HttpMethod.class), any(HttpEntity.class), any(ParameterizedTypeReference.class)))
+                .willThrow(new HttpServerErrorException(INTERNAL_SERVER_ERROR));
+
+        try {
+            hmrcCallWrapper.exchange(new URI("some-uri"), POST, new HttpEntity("some-body"), new ParameterizedTypeReference<Resource<String>>() {
+            });
+        } catch (HttpServerErrorException | URISyntaxException e) {
+
+            then(mockAppender).should(times(1)).doAppend(logArgumentCaptor.capture());
+            LoggingEvent loggingEvent = logArgumentCaptor.getValue();
+            assertThat(loggingEvent.getFormattedMessage()).isEqualTo("Received 500 - INTERNAL_SERVER_ERROR");
+            assertThat(loggingEvent.getArgumentArray()[0]).isEqualTo(HttpStatus.valueOf("INTERNAL_SERVER_ERROR"));
+            assertThat(loggingEvent.getArgumentArray()[1]).isEqualTo("INTERNAL_SERVER_ERROR");
+            assertThat(loggingEvent.getArgumentArray()[2]).isEqualTo(new ObjectAppendingMarker("event_id", HMRC_SERVER_ERROR));
+        }
+    }
+
+    @Test
+    public void shouldLogWhenBadGatewayException() {
+        given(mockRestTemplate.exchange(any(URI.class), any(HttpMethod.class), any(HttpEntity.class), any(ParameterizedTypeReference.class)))
+                .willThrow(new HttpServerErrorException(BAD_GATEWAY));
+
+        try {
+            hmrcCallWrapper.exchange(new URI("some-uri"), POST, new HttpEntity("some-body"), new ParameterizedTypeReference<Resource<String>>() {
+            });
+        } catch (HttpServerErrorException | URISyntaxException e) {
+
+            then(mockAppender).should(times(1)).doAppend(logArgumentCaptor.capture());
+            LoggingEvent loggingEvent = logArgumentCaptor.getValue();
+            assertThat(loggingEvent.getFormattedMessage()).isEqualTo("Received 502 - BAD_GATEWAY");
+            assertThat(loggingEvent.getArgumentArray()[0]).isEqualTo(HttpStatus.valueOf("BAD_GATEWAY"));
+            assertThat(loggingEvent.getArgumentArray()[1]).isEqualTo("BAD_GATEWAY");
```

Comment:
I think asserting on the message content and the event is enough.
`assertThat(loggingEvent.getArgumentArray()[0]).isEqualTo(HttpStatus.valueOf("BAD_GATEWAY"));
            assertThat(loggingEvent.getArgumentArray()[1]).isEqualTo("BAD_GATEWAY");`
can go - they can't break without the line above also breaking.